src/components/bulk-email-tool/bulk-email-form/BulkEmailForm.jsx

```diff
@@ -0,0 +1,221 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Form, Icon, StatefulButton, useCheckboxSetValues, useToggle,
+} from '@edx/paragon';
+import { SpinnerSimple, CheckCircle, Cancel } from '@edx/paragon/icons';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import TextEditor from '../text-editor/TextEditor';
+import { postBulkEmail } from './api';
+import BulkEmailRecepient from './BulkEmailRecepient';
+import TaskAlertModal from './TaskAlertModal';
+import useTimeout from '../../../utils/useTimeout';
+
+export const FORM_SUBMIT_STATES = {
+  DEFAULT: 'default',
+  PENDING: 'pending',
+  COMPLETE: 'complete',
+  COMPLETED_DEFAULT: 'completed_default',
+  ERROR: 'error',
+};
+
+export default function BulkEmailForm(props) {
+  const { courseId } = props;
+  const [subject, setSubject] = useState('');
+  const [emailFormStatus, setEmailFormStatus] = useState(FORM_SUBMIT_STATES.DEFAULT);
+  const [emailformValidation, setEmailFormValidation] = useState({
+    // set these as true on initialization, to prevent invalid messages from prematurely showing
+    subject: true,
+    body: true,
+    recepients: true,
+  });
+  const [selectedRecepients, { add, remove }] = useCheckboxSetValues([]);
+  const [isTastAlertOpen, openTaskAlert, closeTaskAlert] = useToggle(false);
```

Comment:
nit: I think `task` is mispelled in `isTastAlertOpen`.

Reply:
good catch